Lib/ufo2ft/featureWriters/cursFeatureWriter.py

```diff
@@ -0,0 +1,129 @@
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName in dirGlyphs["LTR"]
+                ),
+                direction="LTR",
+            )
+            if LTRlookup:
+                lookups.append(LTRlookup)
+
+            # Make RTL lookup with other glyphs
+            RTLlookup = self._makeCursiveLookup(
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName not in dirGlyphs["LTR"]
+                ),
+                direction="RTL",
+            )
+            if RTLlookup:
+                lookups.append(RTLlookup)
+        else:
+            lookup = self._makeCursiveLookup(
+                (glyph for (glyphName, glyph) in ordereredGlyphSet)
+            )
+            if lookup:
+                lookups.append(lookup)
+
+        if lookups:
+            feature = ast.FeatureBlock("curs")
+            feature.statements.extend(lookups)
+            return feature
+
+    def _makeCursiveLookup(self, glyphs, direction=None):
+        statements = self._makeCursiveStatements(glyphs)
+
+        if not statements:
+            return
+
+        suffix = ""
+        if direction == "LTR":
+            suffix = "_ltr"
+        elif direction == "RTL":
+            suffix = "_rtl"
+        lookup = ast.LookupBlock(name=f"curs{suffix}")
+
+        if direction != "LTR":
+            lookup.statements.append(
+                ast.makeLookupFlag(flags=("IgnoreMarks", "RightToLeft"))
+            )
+        else:
+            lookup.statements.append(ast.makeLookupFlag(name="IgnoreMarks"))
+
+        lookup.statements.extend(statements)
+
+        return lookup
+
+    def _makeCursiveStatements(self, glyphs):
+        cursiveAnchors = dict()
+        statements = []
+        for glyph in glyphs:
+            entryAnchor = exitAnchor = None
+            for anchor in glyph.anchors:
+                if entryAnchor and exitAnchor:
+                    break
+                if anchor.name in self.entryAnchorNames:
+                    entryAnchor = ast.Anchor(x=otRound(anchor.x), y=otRound(anchor.y))
+                elif anchor.name in self.exitAnchorNames:
+                    exitAnchor = ast.Anchor(x=otRound(anchor.x), y=otRound(anchor.y))
+
+            if entryAnchor or exitAnchor:
+                cursiveAnchors[ast.GlyphName(glyph.name)] = (entryAnchor, exitAnchor)
+
+        if cursiveAnchors:
+            for glyphName, anchors in cursiveAnchors.items():
+                statement = ast.CursivePosStatement(glyphName, *anchors)
+                statements.append(statement)
+
+        return statements
+
+    def _write(self):
+        feaFile = self.context.feaFile
+        feature = self._makeCursiveFeature()
+
+        if not feature:
+            return False
+
+        self._insert(feaFile=feaFile, features=[feature])
```

Comment:
Since what RTL means here affects only the lookup flag, it would be safer not to do this script split, since the glyph can potentially be of a LTR script but have the RTL lookup flag.